### PR TITLE
Consistency 🤑

### PR DESCRIPTION
**Pull request by siemenvdn**
For consistency sake, made some adjustments:

- Moved all `ApiResponse` documentation to seperate file
- Moved `cache` from utils to modules, since it has a `.module.ts` file
- Moved all `.module.ts` files to root of folder
- Add sort on documentation Tags and Routes
- Added missing response documentation:
  - `getPermssions`
  - `getApiStatus`
  - `getHealthStatus`
- Functions on controllers are `public (async) functionName`
- Removed `@ApiQuery` decorators in `typesense.controller` and moved added `@ApiProperty` in the QueryDto
- Changed auth-type on controllers to `@ApiBearerAuth()`
- Changed pagination `DEFAULT_OFFSET` to 0
- Move `sentry.ts` to utils folder
- Move `documentation.ts` to config folder
- `/file` route to `/files`
- Add `@Permissions` on every route

### CODE REVIEW

**Comment by jonasbeckers** [comment]
- `@ApiBearerAuth()` should be `@ApiOAuth2([])`
- Can we remove `public` modifiers if they are not needed. I don't see the point if they are public by default.

**Comment by jorenvandeweyer** [comment]
> * Can we remove `public` modifiers if they are not needed. I don't see the point if they are public by default.

I think all calls should require authentication by default.

File: src/modules/permissions/controllers/permission.controller.ts
Comment (PauwelsPieter) [inline src/modules/permissions/controllers/permission.controller.ts:1]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/auth/controllers/auth.controller.ts [hunk 1]
```diff
@@ -33,13 +37,11 @@ export class AuthController {
...
-  @ApiOAuth2([])
-  @ApiResponse({
-    status: 200,
-    description: 'The user info has been successfully retrieved.',
-    type: UserTransformerType
-  })
-  public async getUserInfo (@Req() req: Request): Promise<UserTransformerType> {
+  @ApiBearerAuth()
+  @ApiResponse(getUserInfoResponse)
+  public async getUserInfo (
```
Comment (PauwelsPieter) [inline src/modules/auth/controllers/auth.controller.ts:42]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/files/controllers/file.controller.ts
Comment (PauwelsPieter) [inline src/modules/files/controllers/file.controller.ts:1]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/roles/controllers/role.controller.ts [hunk 2]
```diff
@@ -1,119 +1,88 @@
...
-  })
+  @Post('/:roleUuid')
+  @ApiResponse(updateRoleResponse)
   @Permissions(Permission.ROLE_UPDATE)
-  async updateRole (
-    @Body() updateRoleDto: CreateRoleDto,
-    @Param('role', ParseUUIDPipe) uuid: string
+  public async updateRole (
+    @UuidParam('roleUuid') roleUuid: string,
+    @Body() updateRoleDto: CreateRoleDto
```
Comment (PauwelsPieter) [inline src/modules/roles/controllers/role.controller.ts:64]:
```suggestion
    @Body() updateRoleDto: UpdateRoleDto
```
Aparte type voor update?

File: src/modules/roles/transformers/role-count.transformer.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Transformer } from '@appwise/transformer'
+import { ApiProperty } from '@nestjs/swagger'
+
+export class RoleCountTransformerType {
+  @ApiProperty({ type: Number, description: 'The count of the roles' })
```
Comment (PauwelsPieter) [inline src/modules/roles/transformers/role-count.transformer.ts:5]:
```suggestion
  @ApiProperty({ type: Number, description: 'The count of roles' })
```

File: src/modules/users/controllers/user.controller.ts [hunk 4]
```diff
@@ -33,72 +31,51 @@ export class UserController {
...
-    status: 200,
-    description: 'The user has been successfully received.',
-    type: UserTransformerType
-  })
-  @ApiOAuth2([])
-  @UseGuards(UpdateUserGuard)
+  @Get('/:userUuid')
+  @ApiResponse(getUserResponse)
+  @UseGuards(UserGuard)
   async getUser (
```
Comment (PauwelsPieter) [inline src/modules/users/controllers/user.controller.ts:47]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/users/controllers/user.controller.ts [hunk 5]
```diff
@@ -33,72 +31,51 @@ export class UserController {
...
-    status: 201,
-    description: 'The user has been successfully updated.',
-    type: UserTransformerType
-  })
-  @ApiOAuth2([])
-  @UseGuards(UpdateUserGuard)
+  @Post('/:userUuid')
+  @ApiResponse(updateUserResponse)
+  @UseGuards(UserGuard)
   async updateUser (
```
Comment (PauwelsPieter) [inline src/modules/users/controllers/user.controller.ts:57]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/users/controllers/user.controller.ts [hunk 6]
```diff
@@ -33,72 +31,51 @@ export class UserController {
...
-  @ApiResponse({
-    status: 200,
-    description: 'The user has been successfully deleted.'
-  })
-  @ApiOAuth2([])
-  @UseGuards(UpdateUserGuard)
+  @Delete('/:userUuid')
+  @ApiResponse(deleteUserResponse)
+  @UseGuards(UserGuard)
   async deleteUser (
```
Comment (PauwelsPieter) [inline src/modules/users/controllers/user.controller.ts:68]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/users/controllers/user.controller.ts [hunk 7]
```diff
@@ -33,72 +31,51 @@ export class UserController {
...
-  @ApiResponse({
-    status: 200,
-    description: 'The users password has been successfully updated.'
-  })
-  @ApiOAuth2([])
-  @UseGuards(UpdateUserGuard)
+  @Post('/:userUuid/password')
+  @ApiResponse(updateUserPasswordResponse)
+  @UseGuards(UserGuard)
   async updateUserPassword (
```
Comment (PauwelsPieter) [inline src/modules/users/controllers/user.controller.ts:77]: Permissie afdwingen met `@Permissions` decorator?

File: src/modules/users/docs/user-response.docs.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export const deleteUserResponse: ApiResponseOptions = {
+  status: 200,
+  description: 'The user has been successfully deleted.'
+}
+
+export const updateUserPasswordResponse: ApiResponseOptions = {
+  status: 200,
+  description: 'The users password has been successfully updated.'
```
Comment (PauwelsPieter) [inline src/modules/users/docs/user-response.docs.ts:29]:
```suggestion
  description: 'The user\'s password has been successfully updated.'
```

File: src/modules/users/guards/user.guard.ts [hunk 9]
```diff
@@ -1,10 +1,10 @@
 import { Injectable, type CanActivate, type ExecutionContext } from '@nestjs/common'
 import { Reflector } from '@nestjs/core'
-import { RedisCacheService } from '../../../utils/cache/cache.js'
-import { Permission } from '../../permissions/permission.enum.js'
+import { RedisCacheService } from '../../cache/services/cache.service.js'
+import { Permission } from '../../permissions/enums/permission.enum.js'
 
 @Injectable()
-export class UpdateUserGuard implements CanActivate {
+export class UserGuard implements CanActivate {
```
Comment (PauwelsPieter) [inline src/modules/users/guards/user.guard.ts:7]: `CurrentUserGuard` noemen misschien?

File: src/modules/files/controllers/file.controller.ts
Comment (siemenvdn) [reply src/modules/files/controllers/file.controller.ts:1]: This PR only focuses on structure, and is not intended to change any logic

File: src/modules/users/controllers/user.controller.ts (hunk 4 above)
Comment (siemenvdn) [reply src/modules/users/controllers/user.controller.ts:47]: This PR only focuses on structure, and is not intended to change any logic

File: src/modules/users/controllers/user.controller.ts (hunk 5 above)
Comment (siemenvdn) [reply src/modules/users/controllers/user.controller.ts:57]: This PR only focuses on structure, and is not intended to change any logic

File: src/modules/users/controllers/user.controller.ts (hunk 7 above)
Comment (siemenvdn) [reply src/modules/users/controllers/user.controller.ts:77]: This PR only focuses on structure, and is not intended to change any logic

File: src/modules/users/guards/user.guard.ts (hunk 9 above)
Comment (siemenvdn) [reply src/modules/users/guards/user.guard.ts:7]: That would indicate the existence of a PastUserGuard

File: src/modules/files/controllers/file.controller.ts [hunk 10]
```diff
@@ -26,20 +30,31 @@ export class FileController {
...
+  ): Promise<void> {
+    await this.fileFlowService.remove(fileUuid)
+  }
+
+  @Post('/:fileUuid/confirm-upload')
+  @ApiResponse(confirmFileUploadResponse)
   @HttpCode(200)
-  async confirmFileUpload (
-    @Param('file', ParseUUIDPipe) fileUuid: string
+  @Permissions(Permission.FILE_UPDATE)
```
Comment (PauwelsPieter) [inline src/modules/files/controllers/file.controller.ts:45]:
```suggestion
  @Permissions(Permission.FILE_CREATE)
```

Part of the create flow

**Review by PauwelsPieter — APPROVED** [review]

**Review by nielsstinissen — CHANGES_REQUESTED** [review]

File: src/modules/auth/docs/auth-response.docs.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import { type ApiResponseOptions } from '@nestjs/swagger'
+import { UserTransformerType } from '../../users/transformers/user.transformer.js'
+import { AuthTransformerType } from '../transformers/auth.transformer.js'
+
+export const getUserInfoResponse: ApiResponseOptions = {
```
Comment (Kobe-Kwanten) [inline src/modules/auth/docs/auth-response.docs.ts:5]: geen fan deze splitsing, waarom splits je dit

File: src/modules/auth/controllers/auth.controller.ts [hunk 12]
```diff
@@ -22,27 +24,14 @@ export class AuthController {
...
-  @ApiExtraModels(PasswordGrantBody, RefreshGrantBody)
-  @ApiBody({
-    schema: {
-      oneOf: [
-        { $ref: getSchemaPath(PasswordGrantBody) },
-        { $ref: getSchemaPath(RefreshGrantBody) }
-      ]
-    }
-  })
+  @ApiResponse(createTokenResponse)
```
Comment (Kobe-Kwanten) [inline src/modules/auth/controllers/auth.controller.ts:27]: je kan hier `ApiOkResponse` van maken  maar ook dat is fout want post geeft standaar 201, dus `ApiCreatedResponse`

File: src/modules/auth/docs/auth-response.docs.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  status: 200,
+  description: 'The user info has been successfully retrieved.',
```
Comment (Kobe-Kwanten) [inline src/modules/auth/docs/auth-response.docs.ts:7]: hoe zinvol zijn deze descriptions zelfs?

File: src/modules/auth/docs/auth-response.docs.ts (hunk 11 above)
Comment (Kobe-Kwanten) [inline src/modules/auth/docs/auth-response.docs.ts:5]: constantes liefst als constante benoemen GET_USER_INFO_RESPONSE

File: src/modules/typesense/builder/search-params.builder.ts [hunk 14]
```diff
@@ -4,7 +4,7 @@ import { type SortDirection } from '../../../utils/query/sort-direction.enum.js'
 import { FilterOptions } from '../enums/typesense-filter-options.enum.js'
 
 export const DEFAULT_LIMIT = 10
-export const DEFAULT_OFFSET = 1
+export const DEFAULT_OFFSET = 0
```
[the remark this answers is not in the document]
Comment (Kobe-Kwanten) [reply src/modules/typesense/builder/search-params.builder.ts:7]: idd, opsplitsing van TYPESENSE vs TYPEORM nodig

File: src/utils/params/uuid-param.utiil.ts [hunk 15]
```diff
@@ -0,0 +1,3 @@
+import { Param, ParseUUIDPipe } from '@nestjs/common'
+
+export const UuidParam = (param: string): ParameterDecorator => Param(param, ParseUUIDPipe)
```
Comment (Kobe-Kwanten) [inline src/utils/params/uuid-param.utiil.ts:3]: geen arrow functie als we overal gewone functies gebruiken

**Review by Kobe-Kwanten — CHANGES_REQUESTED** [review]

File: src/modules/auth/docs/auth-response.docs.ts (hunk 11 above)
Comment (siemenvdn) [reply src/modules/auth/docs/auth-response.docs.ts:5]: Controller can get really crowded with decorator stuff if we dont seperate this imho

File: src/modules/auth/docs/auth-response.docs.ts (hunk 13 above)
Comment (siemenvdn) [reply src/modules/auth/docs/auth-response.docs.ts:7]: For now the descriptions are really simple, but for more complex responses i think they serve a purpose. Never enough documentation

File: src/modules/auth/controllers/auth.controller.ts (hunk 12 above)
Comment (siemenvdn) [reply src/modules/auth/controllers/auth.controller.ts]: That's indeed one of the points we needed to discuss, which decorator we're going to use for responses. But I also lean towards what you propose

Comment (siemenvdn) [reply src/modules/auth/controllers/auth.controller.ts]: For now i've changed it to ur proposal 😄

File: src/modules/cache/cache.module.ts [hunk 16]
```diff
@@ -1,12 +1,11 @@
 import { Module } from '@nestjs/common'
 import { RoleRepository } from '../roles/repositories/role.repository.js'
 import { UserRepository } from '../users/repositories/user.repository.js'
-import { CacheService } from '../cache/cache.service.js'
 import { RedisModule } from '../redis/redis.module.js'
+import { CacheService } from './services/cache.service.js'
 
 @Module({
   imports: [
-    // NatsModule.forRoot(),
```
Comment (JonasVannieuwenhuijsen) [inline src/modules/cache/cache.module.ts:9]: Had deze bewust in comment gezet, zodat men weet dat men makkelijk kan switchen tussen Redis of Nats... Maar miss is er een beter manier om dat te documenteren zodat iedere backender dit weet...

File: src/modules/cache/services/cache.service.ts [hunk 17]
```diff
@@ -12,7 +12,6 @@ const userRoleCache = `${prefix}.user-role-cache`
 @Injectable()
 export class CacheService {
   constructor (
-    // private readonly client: NatsClient,
```
Comment (JonasVannieuwenhuijsen) [inline src/modules/cache/services/cache.service.ts:15]: Idem hier, dit zijn de 2 enige dingen die je moet doe om te switchen tussen Redis en Nats

File: src/utils/params/uuid-param.utiil.ts [hunk 18]
```diff
@@ -0,0 +1,5 @@
+import { Param, ParseUUIDPipe } from '@nestjs/common'
+
+export function UuidParam (param: string): ParameterDecorator {
```
Comment (Kobe-Kwanten) [inline src/utils/params/uuid-param.utiil.ts:3]: filename bevat `ii`

File: src/modules/auth/docs/auth-response.docs.ts (hunk 13 above)
Comment (Kobe-Kwanten) [reply src/modules/auth/docs/auth-response.docs.ts:7]: I think they can serve a purpose indeed, but not in these kinds of calls
I definitely don't think you can never have enough documentation. 
Documentation is a burden to keep up to date so I'd only use it where it brings additional value

File: src/modules/auth/docs/auth-response.docs.ts (hunk 11 above)
Comment (Kobe-Kwanten) [reply src/modules/auth/docs/auth-response.docs.ts]: I think you're tackling the wrong issue then. I don't think there are more decorators on a controller call than there need to be. I think there are too many calls in the controller. Hence the vertical slicing approach I suggested.
I'd rather keep information that is closely related, close.

**Review by Kobe-Kwanten — CHANGES_REQUESTED** [review]

**Comment by siemenvdn** [comment]
Going to be splitting this up in smaller changes.

**Pull request closed without merging**